src/redux/features/authDetailsSlice.ts

```diff
@@ -0,0 +1,32 @@
+import { Token } from '@/types/common'
+import { createSlice } from '@reduxjs/toolkit'
+import { RootState } from '@/redux/store'
+
+interface IInitialState {
+  tokenPayload: Token | {}
+  internalUserId: string
+}
+
+const initialState: IInitialState = {
+  tokenPayload: {},
+  internalUserId: '',
```

Comment:
Having `internalUserId` separately is not necessary as we already have `tokenPayload` in this store which consists of the `internalUserId`.

Reply:
Let's not make copies of same data more often because we have to update both of these to maintain consistency if we need updated `tokenPayload` again. Someone may update `tokenPayload` and forget to update `internalUserId`.